tests/testthat/test-rowClust.Rmd

```diff
@@ -0,0 +1,34 @@
+# Loading example data
+
+```{r, message=FALSE, warning=FALSE}
+library(mia)
+data("GlobalPatterns")
+tse <- GlobalPatterns
+tse <- agglomerateByRank(tse, rank="Class")
+source("../../R/rowClust.R")
+```
+
+# Clustering
+
+```{r}
+tse <- rowClust(tse, method = "ward.D2",metric = "spearman", k = 6)
+head(rowData(tse))
+```
+
+```{r}
+tse <- rowClust(tse, method = "average",metric = "cosine", k = 6)
+head(rowData(tse))
+```
+
+# Aggregating by clusters with `mia::mergeRows`
+
+```{r}
+tse_CAG_sum <- mergeRows(tse, rowData(tse)$clust_id)
+head(assay(tse_CAG_sum))
+```
+
+```{r}
+tse_CAG_av <- mergeRows(tse, rowData(tse)$clust_id, average=TRUE)
+head(assay(tse_CAG_av))
+```
```

Comment:
Good solution. But what is `average=TRUE`? I did not find this from `help(mergeRows)`.

By default, I would merge rows by summing up the abundances.

Reply:
It's the arguments passed from `mergeRows` to `sumCountsAcrossFeatures`; help(sumCountsAcrossFeatures).
Ok, should the merging be shown at examples (docs/OMA) or shall it be a separate function?

Reply:
Right. Let us show the merging now as an example in OMA as well as in the manpage of `rowClust` (because it is often needed in that context).